src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -61,6 +66,19 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig, gdDAO: Datap
     }
   }
 
+
+  def processClusterRequest(googleProject: GoogleProject, clusterName: ClusterName, clusterRequest: ClusterRequest): ClusterRequest = {
+    // create a LabelMap of default labels
+    val defaultLabels = DefaultLabels(clusterName, googleProject, clusterRequest.bucketPath, clusterRequest.serviceAccount, clusterRequest.jupyterExtensionUri)
+      .toJson.asJsObject.fields.mapValues(labelValue => labelValue.toString.replaceAll("\"", ""))
+    // combine default and given labels
+    val allLabels = clusterRequest.labels ++ defaultLabels
```

Comment:
What if the user wants to override a default label? Maybe this should be `defaultLabels ++ clusterRequest.labels` to ensure that user-specified labels take precedence.

Reply:
Good point! Do we want the user to be able to override the default labels?

Reply:
Looking at the ticket, it specifies that user labels should not overwrite the defaults.
But I guess the next point of discussion would be whether if the user *does* specify a default label, should we just silently override it with our own, or should it throw an error saying you can't even attempt to input a label with a default label key?

Reply:
Users shouldn't be able to override default labels. Otherwise they'd be able to make a notebook show up in the wrong google project, which would confuse the hell out of Leo and applications that use it.

Reply:
Oh, I think I already said this but it got eaten in a review comment?